servlet/theLearningServlet/src/main/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServlet.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Permission to use, copy, modify, and/or distribute this software for any 
+ * purpose with or without fee is hereby granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR(S) BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+ * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+package jakartaee.examples.servlet.thelearningservlet;
+
+/**
+ * TheLearningServlet Web Application
+ *
+ * Prepared by Ken Fogel omniprof@gmail.com
+ *
+ * Twitter: @omniprof - Blog: www.omnijava.com
+ *
+ */
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet implementation class TheLearningServlet
+ *
+ * The @WebServlet annotation defines this class as a servlet. The description
+ * attribute is informative and this will only appear in the app server's
+ * console. The default name of the servlet is the name of the class but with
+ * the first letter made lowercase. In this class it would be
+ * 'theLearningServlet'. You can change this, as I have, in the attribute
+ * urlPatterns. You can have more than one name for a servlet. This is used when
+ * the name used to access the servlet can act as a selector to determine what
+ * is expected of the servlet as it can discover what name was used to call it.
+ *
+ */
+@WebServlet(urlPatterns = "/learning")
+public class TheLearningServlet extends HttpServlet {
+
+    // All methods log their name just to show the order of calling
+    private static final Logger LOG = Logger.getLogger(TheLearningServlet.class.getName());
+
+    /**
+     * Constructor
```

Comment:
I don't think this is needed. Not only it is logged in the body, but javadoc prepares separate section for constructors.